ladybug_geometry/geometry2d/line.py

```diff
@@ -0,0 +1,224 @@
+        """
+        return LineSegment2D(self.p.rotate(angle, origin), self.v.rotate(angle))
+
+    def reflect(self, normal, origin):
+        """Get a line segment reflected across a plane with the input normal vector and origin.
+
+        Args:
+            normal: A Vector2D representing the normal vector for the plane across
+                which the line segment will be reflected. THIS VECTOR MUST BE NORMALIZED.
+            origin: A Point2D representing the origin from which to reflect.
+        """
+        return LineSegment2D(self.p.reflect(normal, origin), self.v.reflect(normal))
+
+    def scale(self, factor, origin):
+        """Scale a line segment by a factor from an origin point.
+
+        Args:
+            factor: A number representing how much the line segment should be scaled.
+            origin: A Point2D representing the origin from which to scale.
+        """
+        return LineSegment2D(self.p.scale(factor, origin), self.v * factor)
+
+    def scale_world_origin(self, factor):
+        """Scale a line segment by a factor from the world origin. Faster than scale().
+
+        Args:
+            factor: A number representing how much the line segment should be scaled.
+        """
+        return LineSegment2D(self.p.scale_world_origin(factor), self.v * factor)
+
+    def subdivide(self, distances):
+        """Get Point2D values along the line that subdivide it based on input distances.
+
+        Args:
+            distances: A list of distances along the line at which to subdivide it.
+                This can also be a single number that will be repeated unitl the end of
+                the line.
+        """
+        if isinstance(distances, (float, int)):
+            distances = [distances]
+        line_length = self.length
+        dist = distances[0]
+        index = 0
+        sub_pts = [self.p]
+        while dist < line_length:
+            sub_pts.append(self.point_at_length(dist))
+            if index < len(distances) - 1:
+                index += 1
+            dist += distances[index]
+        sub_pts.append(self.p2)
+        return sub_pts
+
+    def subdivide_evenly(self, number):
+        """Get Point2D values along the line that divide it into evenly-spaced segments.
+
+        Args:
+            number: The number of segments into which the line will be divided.
+        """
+        interval = 1 / number
+        parameter = interval
+        sub_pts = [self.p]
+        while parameter < 1:
+            sub_pts.append(self.point_at(parameter))
+            parameter += interval
+        sub_pts.append(self.p2)
+        return sub_pts
+
+    def point_at(self, parameter):
+        """Get a point at a given fraction along the line segment.
+
+        Args:
+            parameter: The fraction between the start and end point where the
+                desired point lies. For example, 0.5 will yield the midpoint.
+        """
+        return self.p + self.v * parameter
```

Comment:
I could do that but you still get a result if the parameter is outside of 0 and 1.  It just lies on the infinite line of the LineSegment3D instead of the line segment itself.  I could imagine some situations where that result might be useful but it's not used at all by any other objects in the library.  I can add it if you think it gives too much freedom and not enough structure.

Reply:
Should be fine as you said and it is explained in docstring.